tests/functions.test.ts

```diff
@@ -41,3 +104,22 @@ describe("FunctionCall", () => {
     expect(user.age).toEqual(30)
   })
 })
+
+describe("FunctionCallValidated", () => {
+  test("Name should be uppercase based on validation check", async () => {
```

Comment:
pulled this down and played around a bit - looks like we are really struggling to get things in uppercase

The only pass I was able to get was to switch over to using MD_JSON - my guess is likely because that mode inserts the schema in a system prompt and includes some instruction

function calls (via tools or otherwise) dont really have any system level instruction at all - maybe we should include something

Reply:
the goal was that the error messages "this should be in uppercase" should prompt the llm to do that, lets try with gpt-4

Reply:
maybe we should change the prompt to not just try again but to fix the erros